src/LiveComponent/src/Test/TestLiveComponent.php

```diff
@@ -0,0 +1,137 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Test;
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadataFactory;
+use Symfony\UX\TwigComponent\ComponentFactory;
+use Symfony\UX\TwigComponent\ComponentMetadata;
+use Symfony\UX\TwigComponent\MountedComponent;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class TestLiveComponent
+{
+    private string $rendered;
+    private array $props;
+    private string $csrfToken;
+    private object $component;
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        private ComponentMetadata $metadata,
+        array $data,
+        private ComponentFactory $factory,
+        private KernelBrowser $client,
+        private LiveComponentHydrator $hydrator,
+        private LiveComponentMetadataFactory $metadataFactory,
+        private UrlGeneratorInterface $router,
+    ) {
+        $this->client->catchExceptions(false);
+
+        $mounted = $this->factory->create($this->metadata->getName(), $data);
+        $props = $this->hydrator->dehydrate(
+            $mounted->getComponent(),
+            $mounted->getAttributes(),
+            $this->metadataFactory->getMetadata($mounted->getName())
+        );
+
+        $this->client->request('GET', $this->router->generate(
+            $this->metadata->get('route'),
+            [
+                '_live_component' => $this->metadata->getName(),
+                'props' => json_encode($props->getProps()),
+            ]
+        ));
+
+        $this->updateState();
+    }
+
+    public function render(): string // todo return RenderedComponent
+    {
+        return $this->rendered;
+    }
+
+    public function component(): object
+    {
+        if (isset($this->component)) {
+            return $this->component;
+        }
+
+        $component = $this->factory->get($this->metadata->getName());
+        $componentAttributes = $this->hydrator->hydrate(
+            $component,
+            $this->props,
+            [],
+            $this->metadataFactory->getMetadata($this->metadata->getName()),
+        );
+
+        return $this->component = (new MountedComponent($this->metadata->getName(), $component, $componentAttributes))->getComponent();
+    }
+
+    /**
+     * @param array<string,mixed> $arguments
+     */
+    public function call(string $action, array $arguments = []): self
+    {
+        return $this->request(['args' => $arguments], $action);
+    }
+
+    /**
+     * @param array<string,mixed> $arguments
+     */
+    public function emit(string $event, array $arguments = []): self
+    {
+        return $this->call($event, $arguments);
```

Comment:
@weaverryan, does any consideration need to be made for emitting events?

Reply:
Hmm, maybe? The point of emitting is to communicate between components. So if ComponentA emits a `Foo` event, then ComponentB might be listening to it.  But are we supporting such multiple-component communication? We would need to keep track of all of the components that are "rendered" currently in order to know if any are listening to `Foo`. Worse, if the component does `emitUp()`, we would need to understand parent-child relationships.

Reply:
Right - that's what I was thinking :). From some quick looking at Livewire, I don't think they do this either. I found a forum somewhere where a user was asserting that one component emitted an event. Then, right after, they were calling the action on a second component that listens to that event. In other words, they were "triggering" the emit on the 2nd component manually - not expecting the system to magically do it.

Reply:
Feels like the best/easiest way to handle this.